packages/iTwinUI-react/src/core/DatePicker/DatePicker.tsx

```diff
@@ -354,10 +472,31 @@ export const DatePicker = (props: DatePickerProps): JSX.Element => {
 
     let dayClass = 'iui-calendar-day';
 
-    if (isSameDay(day, selectedDay)) {
+    const isSelectedDay =
+      isSameDay(day, selectedDay) ||
+      (isSameDay(day, selectedStartDay) && isSameDay(day, selectedEndDay));
+
+    if (isSelectedDay) {
       dayClass += '-selected';
     }
 
+    if (!isSelectedDay && isSameDay(day, selectedStartDay)) {
+      dayClass += '-range-start';
+    }
+
+    if (!isSelectedDay && isSameDay(day, selectedEndDay)) {
+      dayClass += '-range-end';
+    }
```

Comment:
The two new `if` statements could be `else if` statements. Then you could remove the checks for `!isSelectedDay` in the two `else ifs`.

Reply:
If you do this you also could remove the variable and just have it in the one check `if (isSelectedDay) {`, but I like the clarity the variable gives.

Reply:
These `if` statements have been fixed.

Reply:
Why not do the below?
`if (isSelectedDay) {`
`      dayClass += '-selected';`
`    } else if (isSameDay(day, selectedStartDay)) {`
`        dayClass += '-range-start';`
`    } else if (isSameDay(day, selectedEndDay)) {`
`        dayClass += '-range-end';`
`    }`

If a day is the "range-end" and the "range-start" then it's just the "selected" day, right? This way the logic doesn't check for "range-end" if it's "range-start", and looks a bit more streamlined in my opinion.

Reply:
I thought `if else` clauses had to end with an `else` clause, so I didn't want that condition to be empty. It turns out that is not the case, so I've fixed this